### PR TITLE
Resolve BE BUG #577

### PR DESCRIPTION
**Pull request by ttheng3810**
Fix issue #577.

### CODE REVIEW

File: backend/Social/SocialAPI.py [hunk 1]
```diff
@@ -74,15 +74,20 @@ def users():
...
-            "status": "success"
-        }
-        return make_response(response, 400)
+                 "data": list(data),
+                 "status": "success"
+             }
+       
+        if len(userIdList) == 0:
+            return make_response({"err": "userID is not specified", "status": "failed"}), 400 # throws error if userID is not specified in argument
+        for userID in userIdList:
```
Comment (angnobel) [inline backend/Social/SocialAPI.py:83]: 1, Should use a call with $in to complete the task in 1 query instead of making multiple queries

Comment (angnobel) [reply backend/Social/SocialAPI.py:83]: Error only when all userID cannot be found

Comment (angnobel) [reply backend/Social/SocialAPI.py:83]: Should use .count() as well if you only need the number of entries, not the data itself

File: backend/Social/SocialAPI.py [hunk 2]
```diff
@@ -74,15 +74,20 @@ def users():
...
+            if db.Profiles.find_one({"userID": userID}) == None:
+                return make_response({"err": "userID does not exist", "status": "failed"}), 400 # throws error if at least one userID entry in argument does not exist
```
Comment (angnobel) [inline backend/Social/SocialAPI.py:85]: Error code should be 404,  not found

**Review by angnobel — APPROVED** [review]
Refer to comments

**Review by angnobel — CHANGES_REQUESTED** [review]
Accidentally approved, check previous comments

File: backend/Social/SocialAPI.py (hunk 1 above)
Comment (ttheng3810) [reply backend/Social/SocialAPI.py]: Attempt to use ```data.count()``` returns ```'Cursor' object has no attribute 'count'``` error, even though this method is included in [MongoDB documentation](https://docs.mongodb.com/upcoming/reference/method/cursor.count/) (no idea why this is so).

Hence, ```db.Profiles.count_documents({"userID": {'$in': userIdList}}) == 0``` is used as the condition for the non-existence of all userID arguments instead (please refer to latest commit).

**Review by angnobel — APPROVED** [review]
LGTM

**Pull request merged**